exercises/saddle-points/example.rs

```diff
@@ -0,0 +1,24 @@
+
+pub fn find_saddle_points(input: &[&[u64]]) -> Vec<(u64, u64)>{
+    let mut saddle_points: Vec<(u64, u64)> = Vec::new();
+
+    let width = input.len();
+    let height = input[0].len();
+
+    for i in 0..width {
+        for j in 0..height {
+
+            let column = input.iter().map(|x| x[j]).collect::<Vec<u64>>();
+            let row = &input[i];
+            let max = row.iter().max().unwrap();
```

Comment:
> > since this is only based on row, which is only based on i and not j, it would be possible to assign it in the outer loop and not the inner one. Unless that makes the code less understandable.
>
> If the matrix is empty the two for loops prevent an unwrap on row in line 13.

Indeed, if a row is empty. In that case, I will now only optionally ask for `row` to be moved to the outer loop (it still can be), and agree that I was wrong to ask for this `max` line to be moved; it cannot be.

Reply:
Yes you are right. I think in the way it is now it is also more readable.